test/e2e/router_status_test.go

```diff
@@ -0,0 +1,297 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	routev1 "github.com/openshift/api/route/v1"
+
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// TestDeleteIngressControllerShouldClearRouteStatus deletes an Ingress Controller and
+// makes sure a previously admitted route has its status cleared since.
+func TestDeleteIngressControllerShouldClearRouteStatus(t *testing.T) {
+	t.Parallel()
+	// Create an ingress controller that can admit our route.
+	icName := types.NamespacedName{Namespace: operatorNamespace, Name: "ic-delete-test"}
+	domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newPrivateController(icName, domain)
+	ic.Spec.RouteSelector = &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"type": icName.Name,
+		},
+	}
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller: %v", err)
+	}
+	defer assertIngressControllerDeleted(t, kclient, ic)
+
+	// Create a route to be admitted by this ingress controller.
+	// Use openshift-console namespace to get a namespace outside the ingress-operator's cache.
```

Comment:
I had to think a while about why the namespace would matter.  It's because you want to verify that the cleanup logic isn't restricted to routes in the multi-namespace cache, right?

Reply:
Yup exactly. I was once erroneously searching the ingress-operator's cache for routes to clear status on, but routes can be in any namespace (unlike the ingress controllers...). This just verifies the clearing works for routes in all namespaces.